### ISSUE TITLE
Grafolean behind reverse Proxy

### ISSUE THREAD

**Issue by jonasled** [grafolean/grafolean#30]
Hi,
I want grafolean to run behind a traefik reverse proxy. I've configured the grafolean domain to my public domain. When I try to login it tells me wrong password. The password reset button displays the following error: ` CORS not allowed for this origin `

**Comment by grafolean** [comment]
Hi @jonasled,

if I understand correctly, there are two issues at play here?
- the password you have set was not remembered correctly
- the password reset button ("Send password reset e-mail") fails with CORS errors

Needless to say, neither should happen. For the first one, I don't really have a good advice, except to maybe check the logs of the backend to see if something went wrong during the installation.

As for the CORS:
- may I ask where the error is seen? Console or within the web page? Can you post a screenshot? (feel free to hide the domain name and othe information)
- are you maybe using a non-standard port? (there could be a misconfiguration but the system might not catch it correctly - CORS)
- I would suggest trying in a different browser (Firefox has much better error reporting than Chrome in my experience - but it is a good idea to try *both* as they handle errors differently and they could give a hint about the problem)
- checking the logs might also give some clue as to what went wrong: `docker-compose logs`

The idea is that the `Access-Control-Allow-Origin` response headers should match `Origin` request headers (in browser dev tools), so if they don't, cross origin resource sharing will not work.

It is however unusual that this would happen, because there are explicit checks that the domain matches what was configured, so if there is some case when they don't work, I would be very curious about it.

**Comment by jonasled** [comment]
No, I have the correct password, but the login tells me it's wrong.

**Comment by grafolean** [comment]
Sorry if I wasn't clear enough, let me rephrase that:

> the password you have set was not remembered correctly *by Grafolean*

Did you check the logs maybe?

**Comment by grafolean** [comment]
@jonasled Is this still an issue?

For what it's worth, I see no reason why Grafolean would not work behing a reverse proxy, and I haven't seen any of the problems you describe. So if you have some additional information (how to replicate the bug or logs) I would appreciate it. You can also send info through e-mail if you are reluctant to share it over GitHub: anze at grafolean.com.

**Comment by jonasled** [comment]
Sorry for the late answer, the server log has no useful information on login:
```
grafolean                 | 172.29.0.10 - - [02/Oct/2022:20:08:26 +0000] "POST /api/auth/login HTTP/1.1" 403 33 "https://grafolean.xxxxxxxxx/" "Mozilla/5.0 (X11; Linux x86_64; rv:106.0) Gecko/20100101 Firefox/106.0" rt=0.002 urt=0.000
```
If it helps this is my current traefik config for grafolean:
```yaml
      - traefik.http.routers.grafolean-https.rule=Host(`grafolean.xxxxxxxxxxxxxx`)
      - traefik.http.routers.grafolean-https.entrypoints=https
      - traefik.http.routers.grafolean-https.tls=true
      - traefik.http.routers.grafolean-https.tls.certresolver=letsencrypt
      - traefik.http.routers.grafolean-https.middlewares=grafoleanHeader
      - traefik.http.services.grafolean.loadbalancer.server.port=80
      - "traefik.http.middlewares.grafoleanHeader.headers.customrequestheaders.Host=grafolean.xxxxxx
```

**Comment by grafolean** [comment]
Thank you for the info, much appreciated! It is much clearer now.

It is weird though... Looking at the [code](https://github.com/grafolean/grafolean/blob/master/backend/api/auth.py#L21), `/api/auth/login` endpoint should return only `401` (invalid login) or `200`, never `403` (access to resource denied) as the log says. The only reason it would return `403` would be if the authentication middleware incorrectly read the url path and didn't [skip authentication for this endpoint](https://github.com/grafolean/grafolean/blob/master/backend/grafolean.py#L133) (the `/api/auth/login` endpoint takes care of authentication by itself).

This is certainly possible, however, in this case you would also see `Access denied (permissions check failed)` in the logs (loglevel INFO) - do you maybe see it? (the response body would be `Access to resource denied, insufficient permissions` in browser dev tools, if it is easier to check)

In the mean time I will see if I can replicate this setup on some server.

**Comment by grafolean** [comment]
I have tried to reproduce the issue with Traefik, but no luck - it works flawlessly for me. :-/

Do you think you can take a look at the logs again and see if there are any other lines when you try to login?

<details>
<summary>As a reference (and to help anyone else trying to use Grafolean behind Traefik), these are the deployment steps.</summary>

1) I have followed the [first 3 installation steps](https://github.com/grafolean/grafolean#installation)
2) in `docker-compose.yml`:
    a) in service `grafolean` I have commented out `ports:` section (lines 69-70)
    b) in service `grafolean` I have added these lines: (starting at line 82 - careful about indentation)
    ```
        labels:
          - traefik.http.routers.grafolean-https.rule=Host(`grafolean.mydomain.com`)
          - traefik.http.routers.grafolean-https.middlewares=grafoleanHeader
          - traefik.http.services.grafolean.loadbalancer.server.port=80
          - traefik.http.middlewares.grafoleanHeader.headers.customrequestheaders.Host=grafolean.mydomain.com
    ```
    c) I have added service `traefik` at the bottom, after `autoheal` service:
    ```
      traefik:
        image: traefik:v2.9
        container_name: traefik
        restart: always
        # Enables the web UI and tells Traefik to listen to docker
        command: --api.insecure=true --providers.docker
        ports:
          - "${HTTP_PORT:-80}:80"
          # The Web UI (enabled by --api.insecure=true)
          - "8080:8080"
        networks:
          - grafolean
        volumes:
          # So that Traefik can listen to the Docker events
          - /var/run/docker.sock:/var/run/docker.sock
    ```
3) lastly, I have finished the [last 2 installation steps](https://github.com/grafolean/grafolean#installation).
</details>

**Comment by grafolean** [comment]
@jonasled Is this still an issue for you?

Unfortunately I can't reproduce the problem on my side, so the only way I see this could be solved is if you could provide a bigger portion of the logs (sending them privately is also fine - anze at grafolean.com).

**Comment by grafolean** [comment]
Since I can't reproduce the bug and there is not enough information to find it, I will close this issue now. If there really is a bug, I would be happy to hunt it down, so please feel free to reopen the issue if this is the case.
